src/View/Sidebar/VolumeRow.vala

```diff
@@ -0,0 +1,206 @@
+                            uuid = uri;
+                        }
+
+                        list.open_item (this, flag);
+                    }
+                } catch (GLib.Error error) {
+                    // The MountUtil has already shown a dialog on error
+                } finally {
+                    working = false;
+                    add_mountable_tooltip.begin ();
+                }
+            }
+        );
+    }
+
+    private void on_volume_removed (Volume removed_volume) {
+        if (!valid) { //Already removed
+            return;
+        }
+
+        if (volume == removed_volume) {
+            valid = false;
+            list.remove_item_by_id (id);
+        }
+    }
+
+    protected override void on_mount_added (Mount added_mount) {
+        if (added_mount == volume.get_mount ()) {
+            update_visibilities ();
+        }
+    }
+
+    protected override void on_mount_removed (Mount removed_mount) {
+        if (volume.get_mount () == null) {
+            update_visibilities ();
+        }
+    }
+
+    protected override void add_extra_menu_items (PopupMenuBuilder menu_builder) {
+        if (working) {
+            return;
+        }
+
+        add_extra_menu_items_for_mount (volume.get_mount (), menu_builder);
+        add_extra_menu_items_for_drive (volume.get_drive (), menu_builder);
+    }
+
+    protected void add_extra_menu_items_for_drive (Drive? drive, PopupMenuBuilder menu_builder) {
+        if (drive == null) {
+            return;
+        }
+
+        if (drive != null && drive.can_stop ()) {
+            menu_builder
+                .add_separator ()
+                .add_stop_drive (() => { stop_eject_drive.begin (volume.get_drive ()); });
+        } else if (drive.can_eject ()) {
+            menu_builder
+                .add_separator ()
+                .add_eject_drive (() => { stop_eject_drive.begin (volume.get_drive ()); });
+        }
+    }
+
+    protected override async bool get_filesystem_space (Cancellable? update_cancellable) {
+        if (is_mounted) {
+            return yield get_filesystem_space_for_root (volume.get_mount ().get_root (), update_cancellable);
+        } else {
+            return false;
+        }
+    }
+
+    private void open_volume_property_window () {
+        new Files.View.VolumePropertiesWindow (
+            volume.get_mount (),
+            Files.get_active_window ()
+        );
+    }
+
+    protected override void show_mount_info () {
+        if (!is_mounted) {
+            /* Mount the device if possible, defer showing the dialog after
+             * we're done */
+            working = true;
+            Files.FileOperations.mount_volume_full.begin (volume, null, (obj, res) => {
+                try {
+                    Files.FileOperations.mount_volume_full.end (res);
+                } catch (Error e) {
+                } finally {
+                    working = false;
+                }
```

Comment:
What happens, if `working` was already set to true before we are coming to this?

Also, we probably want to print a warning of the error to the console with something like:

```
warning ("Error while showing mount info: %s", e.message);
```

Otherwise it might get hard to spot potential errors around here in the future.

Reply:
This function is called as a callback from a menuitem from `add_extra_menu_items_for_mount ()` in `AbstractMountableRow`, which does check that `working` is not `true`.  So no menuitems are added unless the volume has finished mounting. I can add a `requires` to document this.